pennylane/qnodes/passthru.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+PassthruQNode class
+"""
+import pennylane as qml
+import pennylane.operation
+import pennylane.circuit_graph
+from .base import BaseQNode, QuantumFunctionError
+
+
+"""
+Design notes
+------------
+
+PassthruQNode requires some changes to the way other PennyLane components work:
+
+1. :class:`Operator` must not do domain checking for its parameters, or it must let the ADT pass the check.
+2. The simulator device must return the result as the ADT instead of plain Python/NumPy types.
+3. Any output_conversion in :meth:`BaseQNode.evaluate` must be skipped.
+
+Additionally, any array-like ADT needs to be able to handle (1) scalar multiplication,
+(2) indexing/slicing, and possibly (3) iteration, as these are the things qfuncs expect of
+array-like parameters.
+
+PassthruQNode does not have a Jacobian method, so it does not HAVE to use VariableRefs or scalar linear indexing of input parameters.
+Two options:
+1. Use VariableRefs anyway, re-use most BaseQNode methods.
+   Problem: after evaluating the VariableRefs, stacking sliced/indexed Tensors in Operation.parameters should somehow result in a Tensor, not an object array.
+2. Do not use VariableRefs, call the qfunc each time :meth:`PassthruQNode.evaluate` is called (always mutable).
+   Problem: tensornet_tf requires variable_deps?
```

Comment:
My preference would be option 2. Once default.tensor.tf is ported to QubitDevice, `variable_deps` is no longer needed. Instead, the `apply` method receives the operator objects directly, so the device is able to directly determine the parameter dependencies.

Reply:
Option 2 is what I'm using now :)

Why does the device need to know parameter dependencies? I've forgotten.

Reply:
Originally, so that parametric compilation could be performed in the plugins. However, by passing the operation objects directly, this becomes a lot easier.